opentelemetry-sdk/src/opentelemetry/sdk/_metrics/measurement_consumer.py

```diff
@@ -41,15 +46,31 @@ def collect(
 
 
 class SynchronousMeasurementConsumer(MeasurementConsumer):
+    def __init__(self, sdk_config: SdkConfiguration) -> None:
+        self._lock = Lock()
+        self._sdk_config = sdk_config
+        # should never be mutated
+        self._reader_to_state: Mapping[MetricReader, MetricReaderStorage] = {
+            reader: MetricReaderStorage(sdk_config)
+            for reader in sdk_config.metric_readers
+        }
+        self._async_instruments: List["_Asynchronous"] = []
+
     def consume_measurement(self, measurement: Measurement) -> None:
-        pass
+        for reader_state in self._reader_to_state.values():
+            reader_state.consume_measurement(measurement)
 
     def register_asynchronous_instrument(
         self, instrument: "_Asynchronous"
     ) -> None:
-        pass
+        with self._lock:
+            self._async_instruments.append(instrument)
 
     def collect(
```

Comment:
Just curious, who calls `collect` on the `measurement_consumer`? Does this happen at the end of a metric collection interval (for the asynch instruments)?

Reply:
MetricReader.collect() will be the only thing to call this. So it would be at the end of an interval or in the case of pull metrics (Prometheus), whenever it's scraped.

Reply:
How is `MetricReader.collect()` called? By the exporter?

Reply:
You should give the spec a read through probably from [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader) down. It's supposed to be flexible but there are some examples:

```
+-----------------+            +---------------------------------+
|                 | Metrics... |                                 |
| In-memory state +------------> Periodic exporting MetricReader |
|                 |            |                                 |
+-----------------+            |    +-----------------------+    |
                               |    |                       |    |
                               |    | MetricExporter (push) +-------> Another process
                               |    |                       |    |
                               |    +-----------------------+    |
                               |                                 |
                               +---------------------------------+
```
> One possible implementation of periodic exporting MetricReader is to inherit from MetricReader and start a background task which calls the inherited Collect() method at the requested exportIntervalMillis. The reader's Collect() method may still be invoked by other callers. For example,

and
```
+-----------------+            +-----------------------------+
|                 | Metrics... |                             |
| In-memory state +------------> PrometheusExporter (pull)   +---> Another process (scraper)
|                 |            | (modeled as a MetricReader) |
+-----------------+            |                             |
                               +-----------------------------+
```

> Note: it is expected that the MetricReader.Collect implementations will be provided by the SDK, so it is RECOMMENDED to prevent the user from accidentally overriding it, if possible (e.g. final in C++ and Java, sealed in C#).